buildconfig/pygame-stubs/freetype.pyi

```diff
@@ -60,62 +56,56 @@ class Font:
-        font_index: Optional[int] = 0,
-        resolution: Optional[int] = 0,
-        ucs4: Optional[int] = False,
+        file: Union[str, IO[bytes]],
+        size: Optional[float] = ...,
+        font_index: Optional[int] = ...,
+        resolution: Optional[int] = ...,
+        ucs4: Optional[int] = ...,  # ucs4 is a bool but passed in as an int
     ) -> None: ...
     def get_rect(
-        self,
-        text: str,
-        style: Optional[int] = STYLE_DEFAULT,
-        rotation: Optional[int] = 0,
-        size: Optional[float] = 0,
+        self, text: str, style: Optional[int] = ..., rotation: Optional[int] = ..., size: Optional[float] = ...,
     ) -> Rect: ...
-    def get_metrics(
-        self, text: str, size: Optional[float] = 0
-    ) -> List[Tuple[int, int, int, int, float, float]]: ...
+    def get_metrics(self, text: str, size: Optional[float] = ...) -> Sequence[Tuple[int, int, int, int, float, float]]: ...
     def get_sized_ascender(self, size: float) -> int: ...
     def get_sized_descender(self, size: float) -> int: ...
     def get_sized_height(self, size: float) -> int: ...
     def get_sized_glyph_height(self, size: float) -> int: ...
-    def get_sizes(self) -> List[Tuple[int, int, int, float, float]]: ...
+    def get_sizes(self) -> Sequence[Tuple[int, int, int, float, float]]: ...
     def render(
         self,
         text: str,
-        fgcolor: Optional[_ColorValue] = None,
-        bgcolor: Optional[_ColorValue] = None,
-        style: Optional[int] = STYLE_DEFAULT,
-        rotation: Optional[int] = 0,
-        size: Optional[float] = 0,
+        fgcolor: Optional[_ColorValue] = ...,
+        bgcolor: Optional[_ColorValue] = ...,
+        style: Optional[int] = ...,
+        rotation: Optional[int] = ...,
+        size: Optional[float] = ...,
     ) -> Tuple[Surface, Rect]: ...
     def render_to(
         self,
         surf: Surface,
-        dest: Union[Tuple[int, int], Sequence[int], Rect],
+        dest: Union[Sequence[int], Rect],
         text: str,
-        fgcolor: Optional[_ColorValue] = None,
-        bgcolor: Optional[_ColorValue] = None,
-        style: Optional[int] = STYLE_DEFAULT,
-        rotation: Optional[int] = 0,
-        size: Optional[float] = 0,
+        fgcolor: Optional[_ColorValue] = ...,
+        bgcolor: Optional[_ColorValue] = ...,
+        style: Optional[int] = ...,
+        rotation: Optional[int] = ...,
+        size: Optional[float] = ...,
     ) -> Rect: ...
     def render_raw(
         self,
         text: str,
-        style: Optional[int] = STYLE_DEFAULT,
-        rotation: Optional[int] = 0,
-        size: Optional[float] = 0,
-        invert: Optional[bool] = False,
+        style: Optional[int] = ...,
+        rotation: Optional[int] = ...,
+        size: Optional[float] = ...,
+        invert: Optional[bool] = ...,
     ) -> Tuple[bytes, Tuple[int, int]]: ...
     def render_raw_to(
         self,
-        array: Any,
+        array: BufferProxy,
```

Comment:
I wonder if this accepts memoryview or things like arrays? I think it does, but haven't checked.

The python type system doesn't have support for buffer objects. I opened a thread on it in the types mailing list some months ago, but I don't think there's a good solution.

Reply:
Will do

Reply:
done